Fix missing fields in verbose InfoS

physcat opened the pull request:
Fixes #165

The recent update to v2.1.0 highlighted the omission of `...` in Verbose.InfoS when calling loggingT.infoS.

This complies because the final parameter for loggingT.infoS, accepts `...interface{}` and that interface just thinks all the key value pairs forms a single array. This produces output like `[key value]=(missing)` instead of `key="value"`

physcat commented:
/assign @justinsb

yuzhiquan commented:
Yes, thanks @physcat  very much.
and would you like to add some test for V.InfoS, that is missed.
/cc @serathius

serathius commented:
Would love to add some tests for verbose

serathius commented on klog_test.go:
```diff
@@ -820,6 +820,60 @@ func TestInfoS(t *testing.T) {
+
+			V(l).InfoS(data.msg, data.keysValues...)
+			_, _, line, _ := runtime.Caller(0) //Fragile code, don't move away form the line above
+			line--                             //Calculating the line number of V(l).InfoS statment.
```
```suggestion
			line--                             //Calculating the line number of V(l).InfoS statement.
```

physcat replied:
Darn, the lint got me. Thanks.

serathius commented:
Looks like you changed commit author to one without CLA signed email

physcat commented:
> Looks like you changed commit author to one without CLA signed email

@serathius, yes I was just updating the author on my commit, fixed now.

I've added a verbosity level test `TestVInfoS` based on the original `TestInfoS` tests.

serathius commented on klog_test.go:
```diff
@@ -820,6 +820,60 @@ func TestInfoS(t *testing.T) {
+			logging.file[infoLog] = &flushBuffer{}
+
+			V(l).InfoS(data.msg, data.keysValues...)
+			_, _, line, _ := runtime.Caller(0) //Fragile code, don't move away form the line above
```
Hmm have you considered using `gmtSscanf` to avoid using `runtime.Caller` ? Look at test below `TestErrorS` it doesn't use it.

physcat replied:
I did not find the `scanf` very clear and it gave an odd error when failing to parse the string. But I'm happy to match the current test cases. Consistency is best. The `scanf` has the advantage that it's not fragile in the same way as `runtime.Caller`.

physcat replied:
I've updated the test to better match the existing tests. It does feel a bit wrong to to build the `want` using data from `got`, but it's just the line number and strictly controlled so probably not that bad.

serathius commented:
/lgtm 
/assign @justinsb

serathius commented:
@dims Can you take a look?

dims commented:
/approve
/lgtm
